interface/Tools.h

```diff
@@ -74,6 +76,267 @@ MELAAngles HHAnalyzer::getMELAAngles(const LorentzVector &q1, const LorentzVecto
+    }
+    if (l2_all_indices.empty()) {
+        leptons[dilepton.ilep2].hlt_idx = -1;
+        leptons[dilepton.ilep2].hlt_already_tried_matching = true;
+        if (HH_HLT_DEBUG)
+            std::cout << "\033[31mNo match found for second lepton\033[00m" << std::endl;
+    }
+    // Check that the hlt path name is the same for both legs
+    // FIXME: beware the day of adding single lepton HLT paths....
+    std::vector<int8_t> l1_samepath_indices;
+    std::vector<int8_t> l2_samepath_indices;
+    for (auto& i1: l1_all_indices) {
+        for (auto& i2: l2_all_indices) {
+            if (i1 == i2)
+                continue;
+            for (auto& path1: hlt.object_paths[i1]) {
+                for (auto& path2: hlt.object_paths[i2]) {
+                    if (path1 == path2)
+                    {
+                        l1_samepath_indices.push_back(i1);
+                        l2_samepath_indices.push_back(i2);
+                    }
+                }
+            }
+        }
+    }
+    if (l1_samepath_indices.empty()) {
+        if (HH_HLT_DEBUG)
+            std::cout << "\033[31mNo common HLT name match for the two leptons\033[00m" << std::endl;
+        leptons[dilepton.ilep1].hlt_idx = -1;
+        leptons[dilepton.ilep1].hlt_already_tried_matching = true;
+        leptons[dilepton.ilep1].hlt_DR_matchedObject = std::numeric_limits<float>::max();
+        leptons[dilepton.ilep1].hlt_DPtOverPt_matchedObject = std::numeric_limits<float>::max();
+        leptons[dilepton.ilep2].hlt_idx = -1;
+        leptons[dilepton.ilep2].hlt_already_tried_matching = true;
+        leptons[dilepton.ilep2].hlt_DR_matchedObject = std::numeric_limits<float>::max();
+        leptons[dilepton.ilep2].hlt_DPtOverPt_matchedObject = std::numeric_limits<float>::max();
+        return;
+    }
+    // We have two hlt objects firing the same path: each lepton should be at least leg2, let's make sure of that
+    leptons[dilepton.ilep1].hlt_leg1 = false;
+    leptons[dilepton.ilep1].hlt_leg2 = false;
+    leptons[dilepton.ilep2].hlt_leg1 = false;
+    leptons[dilepton.ilep2].hlt_leg2 = false;
+    // Check who is leg1 who is leg2
+    std::vector<std::string> filter_leg1;
+    std::vector<std::string> filter_leg2;
+    if (leptons[dilepton.ilep1].isMu && leptons[dilepton.ilep2].isMu) {
+        // di-muon filters: from the path name the legs can have asymetric cuts, taken filters from 
+        // https://github.com/cms-analysis/MuonAnalysis-TagAndProbe/blob/fa1f8f3d469a5a78754ed4b4c43adbfad39a2544/python/common_variables_cff.py#L253-L264
+        if (HH_HLT_DEBUG) std::cout << "\tfinding dilepton legs: di-muon" << std::endl;
+        filter_leg1.push_back("hltL3fL1sDoubleMu114L1f0L2f10OneMuL3Filtered17");
+        filter_leg1.push_back("hltL3fL1sDoubleMu114L1f0L2f10L3Filtered17");
+        filter_leg2.push_back("hltL3pfL1sDoubleMu114L1f0L2pf0L3PreFiltered8");
+        filter_leg2.push_back("hltDiMuonGlbFiltered17TrkFiltered8");
```

Comment:
Well if the filter does not exist, you'll never match your leptons and the event will be dropped. This will happen for all events passing `HLT_Mu17_TrkIsoVVL_Mu8_TrkIsoVVL`